daml-lf/verification/utils/Transaction.scala

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package lf.verified
+package utils
+
+import stainless.lang._
+import stainless.annotation._
+import scala.annotation.nowarn
+import stainless.proof._
+import stainless.collection._
+
+import Value.ContractId
+
+/** General transaction type
+  *
+  * Abstracts over NodeId type and ContractId type
+  * ContractId restricts the occurrence of contractIds
+  *
+  * @param nodes The nodes of this transaction.
+  * @param roots References to the root nodes of the transaction.
+  *
+  *              Users of this class may assume that all instances are well-formed, i.e., `isWellFormed.isEmpty`.
+  *              For performance reasons, users are not required to call `isWellFormed`.
+  *              Therefore, it is '''forbidden''' to create ill-formed instances, i.e., instances with `!isWellFormed.isEmpty`.
+  */
+final case class Transaction(
+    nodes: Map[NodeId, Node],
+    roots: List[NodeId],
+)
```

Comment:
Where is this type of tree needed for the verification? Do you actually convert this kind of transaction tree into the `Tree` structure anywhere?